cards-exercise/src/app/app.component.ts

```diff
@@ -1,32 +1,39 @@
 import { Component } from '@angular/core';
+import { Helper } from './helper';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
+  providers: [Helper],
 })
 export class AppComponent {
   title = 'cards-exercise';
+  cards: any[] = [];
 
-  defaultCards = this.generateDefaultCardIds();
+  ngOnInit(): void {}
 
-  getRandomInt() {
-    return Math.floor(Math.random() * 10000);
-  }
+  constructor(private helper: Helper) {}
 
-  sortRandomInts() {
-    this.defaultCards.sort((a, b) => a - b);
+  sortNumber() {
+    this.cards.sort((a, b) => a.number - b.number);
   }
 
   addNewCard() {
-    this.defaultCards.push(Math.floor(this.getRandomInt()));
+    const card = {
+      id: this.helper.createUUID(),
+      number: this.getRandomNumber(),
+    };
+    this.cards.push(card);
   }
 
-  generateDefaultCardIds() {
-    let res = [];
-    for (let i = 0; i < 4; i++) {
-      res.push(Math.floor(this.getRandomInt()));
-    }
-    return res;
+  deleteCard(id: string) {
+    this.cards.forEach((card, index) => {
```

Comment:
use just filter instead of this. You need to avoid editing collection in foreach